### ISSUE TITLE
Integration with Spring Boot 3

### ISSUE THREAD

**Issue by egor-ponomarev** [spring-cloud/spring-cloud-sleuth#2245]
Hello all! 
Currently, it is not possible to use the latest version of Spring Cloud Sleuth 3.1.5 with Spring Boot 3 because of dependency on spring-webmvc 5.3
For example, TraceWebServletConfiguration uses HandlerInterceptorAdapter which is absent in spring webmvc 6.0.
https://github.com/spring-cloud/spring-cloud-sleuth/blob/v3.1.5/spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/web/TraceWebServletConfiguration.java#L66

Does anyone know when the integration with Spring Boot 3 is planned? 

Thank you in advance!

**Comment by marcingrzejszczak** [comment]
We've been mentioning that in multiple places that Sleuth 3.1.x will not work with Boot 3. I've updated the readme, documentation page to reflect that. Also if you try to run cloud 2020.0.0 with Sleuth you will get an exception because of that that tells you how to migrate. We've got at least 3 issues asking about this and numerous stackoverflow questions. We even write about this on this page https://spring.io/projects/spring-cloud-sleuth . Hopefully with the changes in README and docs the message will be easier to be seen
